### PR TITLE
Httpd add possibility to configure custom attributes and resources

### PR DESCRIPTION
**Pull request by TomRoSystems**

### CODE REVIEW

File: instrumentation/httpd/opentelemetry.conf [hunk 1]
```diff
@@ -32,4 +32,7 @@ OpenTelemetryPath /tmp/output-spans
 
 # OpenTelemetryIgnoreInbound off
 
+# OpenTelemetrySetResource service.name apache-web-server
+# OpenTelemetrySetAttribute foo bar
```
Comment (seemk) [inline instrumentation/httpd/opentelemetry.conf:36]: Are these attributes global, for all routes? Is it possible to set attributes for specific route?

With nginx it's possible to set these per server, route basis.

File: instrumentation/httpd/src/otel/opentelemetry.cpp [hunk 2]
```diff
@@ -85,8 +85,17 @@ void initTracer()
     processor = std::make_shared<sdktrace::SimpleSpanProcessor>(std::move(exporter));
   }
 
+  opentelemetry::sdk::resource::ResourceAttributes resAttrs({{"service.name", "httpd"}});
```
Comment (seemk) [inline instrumentation/httpd/src/otel/opentelemetry.cpp:88]: From the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/README.md#service):
```
[1]: MUST be the same for all instances of horizontally scaled services. If the value was not specified, SDKs MUST fallback to u nknown_service: concatenated with process.executable.name, e.g. unknown_service:bash. If process.executable.name is not available, the value MUST be set to unknown_service.
```

File: instrumentation/httpd/src/otel/opentelemetry.cpp [hunk 3]
```diff
@@ -85,8 +85,17 @@ void initTracer()
...
+  // add custom-configured resources
+  for(auto &it:config.resources)
```
Comment (seemk) [inline instrumentation/httpd/src/otel/opentelemetry.cpp:90]: Spacing?

File: instrumentation/httpd/src/otel/opentelemetry.h [hunk 4]
```diff
@@ -54,6 +55,8 @@ struct OtelConfig
   // context propagation
   bool ignore_inbound;
   OtelPropagation propagation;
+  std::unordered_map<std::string, std::string> attribiutes;
```
Comment (seemk) [inline instrumentation/httpd/src/otel/opentelemetry.h:58]: Typo here

**Review by lalitb — APPROVED** [review]

File: instrumentation/httpd/opentelemetry.conf (hunk 1 above)
Comment (TomRoSystems) [reply instrumentation/httpd/opentelemetry.conf:36]: At the moment those are global settings.

File: instrumentation/httpd/src/otel/opentelemetry.cpp (hunk 2 above)
Comment (TomRoSystems) [reply instrumentation/httpd/src/otel/opentelemetry.cpp]: Yes. Thanks. Removed.

File: instrumentation/httpd/src/otel/opentelemetry.h (hunk 4 above)
Comment (TomRoSystems) [reply instrumentation/httpd/src/otel/opentelemetry.h]: Fixed typo.

**Pull request merged**